scrap_reason_code/views/stock_scrap_views.xml

```diff
@@ -7,7 +7,11 @@
         <field name="inherit_id" ref="stock.stock_scrap_form_view" />
         <field name="arch" type="xml">
             <xpath expr="//group/group" position='inside'>
-                <field name="reason_code_id" />
+                <field
+                    name="reason_code_id"
+                    required="1"
+                    domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
```

Comment:
IMHO you should write ir.rule for that. So, no need of domain

Reply:
Domain should be welcome, as on 13.0 we can work on several companies at the same time

Reply:
Indeed, ok!

Reply:
record rules are also added.